torch/distributed/launcher/api.py

```diff
@@ -89,6 +89,12 @@ class LaunchConfig:
     metrics_cfg: Dict[str, str] = field(default_factory=dict)
     local_addr: Optional[str] = None
     local_ranks_filter: Optional[Set[int]] = None
+    # This triggers an alternative file layout for how we put log files into
+    # log_dir; specifically, all log files are prefixed with dedicated_log_
+    # and placed in the top-level directory (no subdirectory structure).
+    # If you think this is generally useful, we can refactor this into some
+    # more generic flags (e.g., log_use_flat_structure and log_name_prefix)
```

Comment:
I think it would be helpful to have some description here of what is the layout is. And if we do that, we might as well have a generic flag?

Or you think that this should be a "do_not_use_fb_layout": we will change this in a BC-breaking way to match fb layout without warning?

Reply:
So, there's two parts to the layout. First, what can you depend on? You can depend that the logs are dropped in a file named `dedicated_log_*` in your log dir. I don't really want to commit to a contract on the rest of the string (but maybe @kurman would say differently, torchelastic has a contract for how its directories are laid out and the filename here matches it.)

I can be convinced we do this as a generic flag. For me, the biggest sticking point is the `dedicated_log_` prefix. This is totally arbitrary and Tupperware-ism, and so I... guess you should let people override the prefix? Or maybe have some sort of generic template format for saying how they want the files to look? I don't want to overbuild this but a template string for the log names is the most logical generalization. If you allow slashes in the string, this could also subsume the directory creation bits too.